install/empty_data.php

```diff
@@ -9479,105 +9492,119 @@ public function getEmptyData(): array
+            } else {
+                throw new LogicException(); // Can't happen.
+            }
 
-            // Assign e2e user all default profiles on the e2e entity
-            $tables['glpi_profiles_users'][] = [
-                'id' => 6,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_SUPER_ADMIN,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 7,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_SELF_SERVICE,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 8,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_OBSERVER,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 9,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_ADMIN,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 10,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_HOTLINER,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 11,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_TECHNICIAN,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
-            ];
-            $tables['glpi_profiles_users'][] = [
-                'id' => 12,
-                'users_id' => 7,
-                'profiles_id' => self::PROFILE_READ_ONLY,
-                'entities_id' => 1,
-                'is_recursive' => 1,
-                'is_dynamic' => 0,
+            // Create required entites
+            foreach ($sub_entities_to_create as $entity) {
+                $subentity1 = array_replace($root_entity, [
+                    'id'           => $next_available_entity_id++,
+                    'name'         => $entity,
+                    'entities_id'  => $e2e_parent_entity_id,
+                    'completename' => __('Root entity')
+                        . " > E2ETestEntity > "
+                        . $entity,
+                    'level'        => 3,
+                ]);
+                $tables['glpi_entities'][] = $subentity1;
+            }
+
+            // Create required users
+            $next_available_user_id = 7;
+            $next_available_profile_id = 6;
```

Comment:
It should be computed with `max(array_column($tables['glpi_XXX'], 'id'));` to be more resilient to changes.